### PR TITLE
Feat/add internationalization select multiple component

### PR DESCRIPTION
**Pull request by tol-any**
#### What does this PR do?

Fixing the Issue https://github.com/grommet/grommet/issues/6337

It's a follow-up to https://github.com/grommet/grommet/pull/6655#issuecomment-1668688234

### CODE REVIEW

File: src/js/components/Grommet/index.d.ts [hunk 1]
```diff
@@ -113,6 +113,23 @@ export interface GrommetProps {
...
         selected?: string;
       };
+      selectMultiple?: {
+        multiple?: string;
+        clearAllTitle?: string;
+        selectAllTitle?: string;
+        selectedMultipleNonTotal?: string;
+        selectedMultiple?: string;
+        onMore?: string;
+        //accessability
```
Comment (britt6612) [inline src/js/components/Grommet/index.d.ts:123]: should this be taken out?

File: src/js/languages/default.json [hunk 2]
```diff
@@ -91,6 +91,22 @@
     "multiple": "multiple",
     "selected": "; Selected: {currentSelectedValue}"
   },
+  "selectMultiple": {
+    "multiple": "{selectedCount} of {totalCount} selected",
+    "clearAllTitle": "Clear All",
+    "selectAllTitle": "Select All",
+    "selectedMultipleNonTotal": "{selectedCount} selected",
+    "selectedMultiple": "{selectedCount} of {totalCount} selected",
+    "onMore": "+ {showMoreCount} more",
```
Comment (britt6612) [inline src/js/languages/default.json:100]:
```suggestion
    "onMore": "+ {showMoreCount} more options",
```

File: src/js/components/SelectMultiple/index.d.ts [hunk 3]
```diff
@@ -8,6 +8,23 @@ export interface SelectMultipleProps extends BasicSelectProps {
...
   showSelectedInline?: boolean;
   sortSelectedOnClose?: boolean;
+  messages?: {
+    multiple?: string;
+    clearAllTitle?: string;
+    selectAllTitle?: string;
+    selectedMultipleNonTotal?: string;
+    selectedMultiple?: string;
+    onMore?: string;
+    //accessability
```
Comment (britt6612) [inline src/js/components/SelectMultiple/index.d.ts:18]: remove?

**Comment by tol-any** [comment]
Hey, thanks for your help @britt6612  @jcfilben :), I struggle with the snapshot test I really dont know when and how to update theme cleanly...

**Comment by tol-any** [comment]
Good news, the tests are fixed, let me know if there's something else to do

**Comment by britt6612** [comment]
> fix: snapshots

Great we will plan on reviewing this week! Glad you got the tests fixed

File: src/js/components/Select/propTypes.js [hunk 4]
```diff
@@ -106,6 +103,9 @@ if (process.env.NODE_ENV !== 'production') {
         ]),
       ),
     ]),
+    messages: PropTypes.shape({
+      multiple: PropTypes.string,
+    }),
     multiple: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (tollr) [reply src/js/components/Select/propTypes.js:109]: The `messages` key then go in the `genericSelectProps` ? I'll commt it right away but I wanted to check with you

File: src/js/components/SelectMultiple/index.d.ts [hunk 5]
```diff
@@ -8,6 +8,22 @@ export interface SelectMultipleProps extends BasicSelectProps {
...
+    selectedMultiple?: string;
+    onMore?: string;
+    clearAll?: string;
+    selectAll?: string;
+    selectDropDown?: string;
+    dropDown?: string;
+    searchFilter?: string;
+    optionSelected?: string;
+    optionNotSelected?: string;
+    selectedOptions?: string;
```
[the remark this answers is not in the document]
Comment (tollr) [reply src/js/components/SelectMultiple/index.d.ts:25]: Did you have specific namings in line ? I thought about these

| Current | Ideas |
|--------|--------|
| multiple | multipleSelectedLabel |
| selectedMultipleNonTotal | selectedCountLabel |
| selectedMultiple | currentCountOnTotalSelectedLabel |

File: src/js/components/Select/propTypes.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (tol-any) [reply src/js/components/Select/propTypes.js]: Corrected with the last commit ;)

**Comment by tol-any** [comment]
Sorry about the delay, I pushed two commits in response of the last comments. I'd need support for the naming of the keys (https://github.com/grommet/grommet/pull/7009#discussion_r1411267911), I'm not not a native speaker and I have little (to no) knowledge in your namings conventions but I'd be happy to discuss it with you :).

(tollr is my personal github account btw)

File: src/js/components/SelectMultiple/index.d.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/components/SelectMultiple/index.d.ts:25]: Hey @tollr, we were able to discuss as a team about some naming suggestions for the keys and the variable names in the values. I'll comment them as suggestions in the file.

File: src/js/languages/default.json [hunk 6]
```diff
@@ -94,6 +94,22 @@
     "multiple": "multiple",
     "selected": "; Selected: {currentSelectedValue}"
   },
+  "selectMultiple": {
+    "multiple": "{selectedCount} of {totalCount} selected",
```
Comment (taysea) [inline src/js/languages/default.json:98]:
```suggestion
    "summarizedValue": "{selected} of {total} selected",
```

File: src/js/languages/default.json [hunk 7]
```diff
@@ -94,6 +94,22 @@
...
+    "clearAllTitle": "Clear All",
```
Comment (taysea) [inline src/js/languages/default.json:99]:
```suggestion
    "clearAll": "Clear All",
```

File: src/js/languages/default.json [hunk 8]
```diff
@@ -94,6 +94,22 @@
...
+    "selectAllTitle": "Select All",
```
Comment (taysea) [inline src/js/languages/default.json:100]:
```suggestion
    "selectAll": "Select All",
```

File: src/js/languages/default.json [hunk 9]
```diff
@@ -94,6 +94,22 @@
...
+    "selectedMultipleNonTotal": "{selectedCount} selected",
```
Comment (taysea) [inline src/js/languages/default.json:101]:
```suggestion
    "selected": "{selected} selected",
```

File: src/js/languages/default.json (hunk 6 above)
Comment (taysea) [reply src/js/languages/default.json:98]: These suggestions with changes for the value will also require changes in the component files to make sure the variable section is passed through properly.

File: src/js/languages/default.json [hunk 10]
```diff
@@ -94,6 +94,22 @@
...
+    "selectedMultiple": "{selectedCount} of {totalCount} selected",
```
Comment (taysea) [inline src/js/languages/default.json:102]:
```suggestion
    "selectedOfTotal": "{selected} of {total} selected",
```

File: src/js/languages/default.json [hunk 11]
```diff
@@ -94,6 +94,22 @@
...
+    "onMore": "+ {showMoreCount} more options",
+    "clearAll": "{selectedCount} options selected. Clear all?",
```
Comment (taysea) [inline src/js/languages/default.json:104]:
```suggestion
    "clearAllA11y": "{selected} options selected. Clear all?",
```

File: src/js/languages/default.json [hunk 12]
```diff
@@ -94,6 +94,22 @@
...
+    "selectAll": "Select all {totalCount} options",
```
Comment (taysea) [inline src/js/languages/default.json:105]:
```suggestion
    "selectAllA11y": "Select all {total} options",
```

File: src/js/languages/default.json [hunk 13]
```diff
@@ -94,6 +94,22 @@
...
+    "selectDropDown": "Select dropdown",
+    "dropDown": "Open Drop",
```
Comment (taysea) [inline src/js/languages/default.json:107]:
```suggestion
    "open": "Open drop",
```

File: src/js/languages/default.json [hunk 14]
```diff
@@ -94,6 +94,22 @@
...
+    "searchFilter": "Search to filter options",
```
Comment (taysea) [inline src/js/languages/default.json:108]:
```suggestion
    "search": "Search to filter options",
```

**Review by taysea — COMMENTED** [review]
Dropped a few naming suggestions!

**Comment by tol-any** [comment]
Just pushed a bunch of commits to update the messages keys :)

Waiting on further comments!

File: src/js/components/Grommet/index.d.ts [hunk 15]
```diff
@@ -113,6 +113,22 @@ export interface GrommetProps {
         multiple?: string;
         selected?: string;
       };
+      selectMultiple?: {
+        summarizedValue?: string;
```
Comment (britt6612) [inline src/js/components/Grommet/index.d.ts:117]:
```suggestion

```

File: src/js/components/Grommet/index.d.ts [hunk 16]
```diff
@@ -113,6 +113,22 @@ export interface GrommetProps {
...
+        clearAll?: string;
+        selectAll?: string;
+        selected?: string;
```
Comment (britt6612) [inline src/js/components/Grommet/index.d.ts:120]:
```suggestion
        multiple?: string;
```

File: src/js/components/Grommet/index.d.ts [hunk 17]
```diff
@@ -113,6 +113,22 @@ export interface GrommetProps {
...
+        selectedOfTotal?: string;
```
Comment (britt6612) [inline src/js/components/Grommet/index.d.ts:121]:
```suggestion
        onMore?: string;
```

File: src/js/components/Grommet/index.d.ts (hunk 16 above)
Comment (britt6612) [reply src/js/components/Grommet/index.d.ts:120]: we were missing `multple` here

File: src/js/languages/default.json [hunk 18]
```diff
@@ -93,6 +93,22 @@
     "multiple": "multiple",
     "selected": "; Selected: {currentSelectedValue}"
   },
+  "selectMultiple": {
+    "summarizedValue": "{selected} of {total} selected",
+    "clearAll": "Clear All",
+    "selectAll": "Select All",
```
Comment (britt6612) [inline src/js/languages/default.json:99]:
```suggestion
    "clearAllA11y": "{selected} options selected. Clear all?",
```

File: src/js/languages/default.json [hunk 19]
```diff
@@ -93,6 +93,22 @@
...
+    "selected": "{selected} selected",
+    "selectedOfTotal": "{selected} of {total} selected",
+    "onMore": "+ {showMoreCount} more options",
```
Comment (britt6612) [inline src/js/languages/default.json:102]:
```suggestion
    "open": "Open Drop",
```

File: src/js/languages/default.json [hunk 20]
```diff
@@ -93,6 +93,22 @@
...
+    "clearAllA11y": "{selected} options selected. Clear all?",
+    "selectAllA11y": "Select all {total} options",
```
Comment (britt6612) [inline src/js/languages/default.json:104]:
```suggestion
    "optionSelected": "{optionLabel} selected",
```

File: src/js/languages/default.json [hunk 21]
```diff
@@ -93,6 +93,22 @@
...
+    "selectDropDown": "Select dropdown",
```
Comment (britt6612) [inline src/js/languages/default.json:105]:
```suggestion
    "search": "Search to filter options",
```

File: src/js/languages/default.json [hunk 22]
```diff
@@ -93,6 +93,22 @@
...
+    "open": "Open Drop",
```
Comment (britt6612) [inline src/js/languages/default.json:106]:
```suggestion
    "selectAll": "Select All",
```

File: src/js/languages/default.json [hunk 23]
```diff
@@ -93,6 +93,22 @@
...
+    "search": "Search to filter options",
```
Comment (britt6612) [inline src/js/languages/default.json:107]:
```suggestion
    "selectAllA11y": "Select all {total} options",
```

File: src/js/languages/default.json [hunk 24]
```diff
@@ -93,6 +93,22 @@
...
+    "optionSelected": "{optionLabel} selected",
```
Comment (britt6612) [inline src/js/languages/default.json:108]:
```suggestion
    "selectDropDown": "Select dropdown",
```

File: src/js/languages/default.json [hunk 25]
```diff
@@ -93,6 +93,22 @@
...
+    "optionNotSelected": "{optionLabel} not selected",
```
Comment (britt6612) [inline src/js/languages/default.json:109]:
```suggestion
    "selected": "{selected} selected",
```

File: src/js/languages/default.json [hunk 26]
```diff
@@ -93,6 +93,22 @@
...
+    "selectedOptions": "Selected Options"
```
Comment (britt6612) [inline src/js/languages/default.json:110]:
```suggestion
     "selectedOfTotal": "{selected} of {total} selected",
    "selectedOptions": "Selected Options",
    "summarizedValue": "{selected} of {total} selected"
```

File: src/js/components/SelectMultiple/index.d.ts [hunk 27]
```diff
@@ -8,6 +8,22 @@ export interface SelectMultipleProps extends BasicSelectProps {
   value?: (string | number | object)[];
   showSelectedInline?: boolean;
   sortSelectedOnClose?: boolean;
+  messages?: {
```
Comment (britt6612) [inline src/js/components/SelectMultiple/index.d.ts:11]: If you can also list these alphabetical

```
  messagess?: {
    clearAll?: string;
    clearAllA11y?: string;
    multiple?: string;
    onMore?: string;
    open?: string;
    optionNotSelected?: string;
    optionSelected?: string;
    search?: string;
    selectAll?: string;
    selectAllA11y?: string;
    selectDropDown?: string;
    selected?: string;
    selectedOfTotal?: string;
    selectedOptions?: string;
    summarizedValue?: string;
  };

```

File: src/js/components/SelectMultiple/propTypes.js [hunk 28]
```diff
@@ -14,6 +14,22 @@ if (process.env.NODE_ENV !== 'production') {
     ),
     help: PropTypes.node,
     limit: PropTypes.number,
+    messages: PropTypes.shape({
```
Comment (britt6612) [inline src/js/components/SelectMultiple/propTypes.js:17]: also put these in alphabetical order please 

```
    messages: PropTypes.shape({
      clearAll: PropTypes.string,
      clearAllA11y: PropTypes.string,
      multiple: PropTypes.string,
      onMore: PropTypes.string,
      open: PropTypes.string,
      optionNotSelected: PropTypes.string,
      optionSelected: PropTypes.string,
      search: PropTypes.string,
      selectAll: PropTypes.string,
      selectAllA11y: PropTypes.string,
      selectDropDown: PropTypes.string,
      selected: PropTypes.string,
      selectedOfTotal: PropTypes.string,
      selectedOptions: PropTypes.string,
      summarizedValue: PropType.string,
    }),

```

**Review by britt6612 — COMMENTED** [review]
Just one final request thank you for your patience with us :)

File: src/js/components/Grommet/index.d.ts (hunk 16 above)
Comment (tol-any) [reply src/js/components/Grommet/index.d.ts]: a missing renaming I think 🤔 ? 

https://github.com/grommet/grommet/pull/7009/commits/f284136212498f48bd7149062b88dae47f71ac17

let me know if that's okay!

**Comment by tol-any** [comment]
I think we're good on the alphabetical order :) I left one thread open to be sure we agree on it.

>  thank you for your patience with us :)

you're very welcome, I'm learning a lot !

File: src/js/components/Grommet/index.d.ts (hunk 16 above)
Comment (britt6612) [reply src/js/components/Grommet/index.d.ts]: you are correct thank you for clarifying

**Comment by britt6612** [comment]
Just one more file left to organize which is `Grommet/propTypes.js`

```
    messages: PropTypes.shape({
      clearAll: PropTypes.string,
      clearAllA11y: PropTypes.string,
      onMore: PropTypes.string,
      open: PropTypes.string,
      optionNotSelected: PropTypes.string,
      optionSelected: PropTypes.string,
      search: PropTypes.string,
      selectAll: PropTypes.string,
      selectAllA11y: PropTypes.string,
      selectDropDown: PropTypes.string,
      selected: PropTypes.string,
      selectedOfTotal: PropTypes.string,
      selectedOptions: PropTypes.string,
      summarizedValue: PropTypes.string,
    }),
```

**Review by britt6612 — APPROVED** [review]
LGTM

**Review by jcfilben — COMMENTED** [review]
On the `showSelectedInline` version of SelectMultiple I am no longer seeing the 'x of y selected' message when I have options selected and the drop is closed. See the SelectMultiple/showSelectedInline storybook example.

Screen Shot 2024-01-26 at 12 07 26 PM

Screen Shot 2024-01-26 at 12 10 07 PM

File: src/js/languages/default.json [hunk 29]
```diff
@@ -93,6 +93,22 @@
...
     "selected": "; Selected: {currentSelectedValue}"
   },
+  "selectMultiple": {
+    "summarizedValue": "{selected} of {total} selected",
+    "clearAll": "Clear All",
+    "selectAll": "Select All",
+    "selected": "{selected} selected",
+    "selectedOfTotal": "{selected} of {total} selected",
+    "onMore": "+ {showMoreCount} more options",
+    "clearAllA11y": "{selected} options selected. Clear all?",
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/languages/default.json:98]: I'm not following this suggestion @britt6612 . Is this still relevant?

File: src/js/languages/default.json (hunk 22 above)
Comment (taysea) [reply src/js/languages/default.json:100]: I'm not following this suggestion @britt6612 . Is this still relevant?

Comment (britt6612) [reply src/js/languages/default.json:100]: no longer relevant

File: src/js/languages/default.json (hunk 29 above)
[the remark this answers is not in the document]
Comment (britt6612) [reply src/js/languages/default.json:98]: no it was for the alphabetical order but he fixed that already

File: src/js/components/SelectMultiple/SelectionSummary.js [hunk 30]
```diff
@@ -68,13 +71,19 @@ const SelectionSummary = ({
...
     (search !== '' && search !== undefined)
-      ? `${value?.length || 0} selected`
-      : `${value?.length || 0} selected of ${options.length}`;
+      ? 'selectMultiple.selected'
+      : 'selectMultiple.selectedOfTotal';
+
+  const summaryText = format({
+    id: messageId,
+    messages,
+    values: { selected: value?.length, total: options.length },
```
[the remark this answers is not in the document]
Comment (tol-any) [reply src/js/components/SelectMultiple/SelectionSummary.js:85]: I'll try to fix it today!

[the remark this answers is not in the document]
Comment (tol-any) [reply src/js/components/SelectMultiple/SelectionSummary.js]: I missed an update of locale, but I think we're good now!

**Comment by taysea** [comment]
Thanks so much for the work here @tol-any. As I reviewed, a few more refinements came up. I didn't want them to get lost in review comments, so for clarity I've filed a separate PR which branches off of yours. The latest commit includes my refinements: https://github.com/grommet/grommet/pull/7104/commits/df0d7be8c21b03b53b35838d10dcd75edb62550a

I couldn't tag you as a reviewer, but would you like to take a look to make sure all looks good? I've tagged others for final review as well.

**Pull request closed without merging**